exchanges/okx/okx_websocket.go

```diff
@@ -396,84 +423,111 @@ func (ok *Okx) handleSubscription(operation string, subscriptions subscription.L
+			}
+		}
+		switch s.Channel {
+		case okxChannelInstruments, okxChannelPositions, okxChannelOrders, okxChannelAlgoOrders,
+			okxChannelAlgoAdvance, okxChannelLiquidationWarning, okxChannelSpotGridOrder,
+			okxChannelGridOrdersContract, okxChannelMoonGridAlgoOrders, okxChannelEstimatedPrice,
+			okxADLWarning, okxLiquidationOrders, okxRecurringBuyChannel, okxCopyTrading:
+			instrumentType = ok.GetInstrumentTypeFromAssetItem(subscriptions[i].Asset)
+		}
+		switch s.Channel {
+		case okxChannelOptionTrades, okxADLWarning:
+			instrumentFamily, _ = subscriptions[i].Params["instFamily"].(string)
+		}
+
+		switch s.Channel {
+		case okxChannelPositions, okxChannelOrders, okxChannelAlgoOrders,
+			okxChannelEstimatedPrice, okxChannelOptSummary:
+			var underlying string
+			for p := range s.Pairs {
+				underlying, _ = ok.GetUnderlying(s.Pairs[p], subscriptions[i].Asset)
+				underlyings = append(underlyings, underlying)
 			}
 		}
-		if arg.Channel == okxChannelInstruments ||
-			arg.Channel == okxChannelPositions ||
-			arg.Channel == okxChannelOrders ||
-			arg.Channel == okxChannelAlgoOrders ||
-			arg.Channel == okxChannelAlgoAdvance ||
-			arg.Channel == okxChannelLiquidationWarning ||
-			arg.Channel == okxChannelSpotGridOrder ||
-			arg.Channel == okxChannelGridOrdersContract ||
-			arg.Channel == okxChannelEstimatedPrice {
-			instrumentType = ok.GetInstrumentTypeFromAssetItem(s.Asset)
-		}
-
-		if arg.Channel == okxChannelPositions ||
-			arg.Channel == okxChannelOrders ||
-			arg.Channel == okxChannelAlgoOrders ||
-			arg.Channel == okxChannelEstimatedPrice ||
-			arg.Channel == okxChannelOptSummary {
-			if len(s.Pairs) == 1 {
-				underlying, _ = ok.GetUnderlying(s.Pairs[0], s.Asset)
+		args := []SubscriptionInfo{}
+		if len(instrumentIDs) > 0 {
+			for iid := range instrumentIDs {
+				args = append(args, SubscriptionInfo{
+					Channel:          s.Channel,
+					InstrumentType:   instrumentType,
+					UID:              uid,
+					AlgoID:           algoID,
+					InstrumentID:     instrumentIDs[iid],
+					InstrumentFamily: instrumentFamily,
+				})
+			}
+			if len(underlyings) > 0 {
+				if len(underlyings) != len(instrumentIDs) {
+					return errors.New("mismatch in instrument IDs and underlyings length")
+				}
+				for uliID := range underlyings {
+					args[uliID].Underlying = underlyings[uliID]
+				}
 			}
+		} else {
+			args = append(args, SubscriptionInfo{
+				Channel:          s.Channel,
+				InstrumentType:   instrumentType,
+				UID:              uid,
+				AlgoID:           algoID,
+				InstrumentFamily: instrumentFamily,
+			})
 		}
-		arg.InstrumentID = instrumentID
-		arg.Underlying = underlying
-		arg.InstrumentType = instrumentType
-		arg.UID = uid
-		arg.AlgoID = algoID
 
 		if authSubscription {
 			authChannels = append(authChannels, s)
-			authRequests.Arguments = append(authRequests.Arguments, arg)
+			authRequests.Arguments = append(authRequests.Arguments, args...)
```

Comment:
I think that by changing from a single arg appended each time, to `args...` you break the logic on L535 which assumes that you can just snip off the last argument to fit back within `maxConnByteLen`
You'd probably need to make the check below a for loop.
I'm also concerned about what would happen if somehow a single request was too big.

I'll be offering up a PR to add sub conf to this branch, and I may be able to fix this at the same time